pkg/agent/controller/networkpolicy/reject_test.go

```diff
@@ -0,0 +1,277 @@
+		hasTun        bool
+		srcInterface  *interfacestore.InterfaceConfig
+		dstInterface  *interfacestore.InterfaceConfig
+		expectInPort  uint32
+		expectOutPort uint32
+	}{
+		{
+			name:          "RejectPodLocal",
+			rejectType:    RejectPodLocal,
+			srcInterface:  sIface,
+			dstInterface:  dIface,
+			expectInPort:  uint32(sIface.OFPort),
+			expectOutPort: uint32(dIface.OFPort),
+		},
+		{
+			name:          "RejectPodLocalToRemote",
+			rejectType:    RejectPodLocalToRemote,
+			srcInterface:  sIface,
+			expectInPort:  uint32(sIface.OFPort),
+			expectOutPort: 0,
+		},
+		{
+			name:          "RejectPodLocalToRemoteExternal",
+			rejectType:    RejectPodLocalToRemote,
+			srcInterface:  externalSIface,
+			expectInPort:  uint32(externalSIface.OFPort),
+			expectOutPort: uint32(externalSIface.UplinkPort.OFPort),
+		},
+		{
+			name:          "RejectPodRemoteToLocal",
+			rejectType:    RejectPodRemoteToLocal,
+			dstInterface:  dIface,
+			expectInPort:  uint32(gwPort),
+			expectOutPort: uint32(dIface.OFPort),
+		},
+		{
+			name:          "RejectPodRemoteToLocalExternal",
+			rejectType:    RejectPodRemoteToLocal,
+			dstInterface:  externalDIface,
+			expectInPort:  uint32(externalDIface.UplinkPort.OFPort),
+			expectOutPort: uint32(externalDIface.OFPort),
+		},
+		{
+			name:          "RejectServiceLocal",
+			rejectType:    RejectServiceLocal,
+			srcInterface:  sIface,
+			expectInPort:  uint32(sIface.OFPort),
+			expectOutPort: uint32(0),
+		},
+		{
+			name:          "RejectServiceLocalToRemote",
+			rejectType:    RejectServiceLocalToRemote,
+			srcInterface:  sIface,
+			expectInPort:  uint32(sIface.OFPort),
+			expectOutPort: uint32(0),
+		},
+		{
+			name:          "RejectServiceRemoteToLocal",
+			rejectType:    RejectServiceRemoteToLocal,
+			expectInPort:  uint32(gwPort),
+			expectOutPort: uint32(0),
+		},
+		{
+			name:          "RejectNoAPServiceLocal",
+			rejectType:    RejectNoAPServiceLocal,
+			srcInterface:  sIface,
+			expectInPort:  uint32(sIface.OFPort),
+			expectOutPort: uint32(gwPort),
+		},
+		{
+			name:          "RejectNoAPServiceRemoteToLocal",
+			rejectType:    RejectNoAPServiceRemoteToLocal,
+			hasTun:        true,
+			expectInPort:  uint32(1),
+			expectOutPort: uint32(gwPort),
+		},
+		{
+			name:          "RejectNoAPServiceRemoteToLocalWithoutTun",
+			rejectType:    RejectNoAPServiceRemoteToLocal,
+			hasTun:        false,
+			expectInPort:  uint32(gwPort),
+			expectOutPort: uint32(gwPort),
+		},
+		{
+			name:          "RejectServiceRemoteToExternal",
+			rejectType:    RejectServiceRemoteToExternal,
+			hasTun:        true,
+			expectInPort:  uint32(1),
+			expectOutPort: uint32(0),
+		},
+		{
+			name:          "RejectServiceRemoteToExternalWithoutTun",
+			rejectType:    RejectServiceRemoteToExternal,
+			hasTun:        false,
+			expectInPort:  uint32(gwPort),
+			expectOutPort: uint32(0),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tunPort := uint32(0)
```

Comment:
You can move this out of for conditions and also reuse it in the test structure instead of `uint32(0)`